File: java/src/main/java/com/github/AndrewAlbizati/Bot.java
package com.github.AndrewAlbizati;

import com.github.AndrewAlbizati.commands.Ping;
import org.javacord.api.DiscordApi;
import org.javacord.api.DiscordApiBuilder;
import org.javacord.api.interaction.SlashCommand;

public class Bot {
    private final String token;
    private DiscordApi api;

    public Bot(String token) {
        this.token = token;
    }

    public DiscordApi getApi() {
        return api;
    }

    /**
     * Starts the Discord bot and initializes Blackjack and Help commands
     */
    public void run() {
        // Start Discord bot
        api = new DiscordApiBuilder().setToken(token).login().join();
        System.out.println("Logged in as " + api.getYourself().getDiscriminatedName());

        // Create slash command (may take a few minutes to update on Discord)
        SlashCommand.with("ping", "Pings the bot.").createGlobal(api).join();

        // Create slash command listener
        api.addSlashCommandCreateListener(new Ping(this));
    }
}

File: kotlin/src/main/kotlin/com/github/andrewalbizati/Bot.kt
package com.github.andrewalbizati

import com.github.andrewalbizati.commands.Ping
import org.javacord.api.DiscordApi
import org.javacord.api.DiscordApiBuilder
import org.javacord.api.interaction.SlashCommand

class Bot(initialToken:String) {
    lateinit var api : DiscordApi
        private set

    private val token : String
    init {
        token = initialToken
    }

    /**
     * Starts the Discord bot and initializes commands
     */
    fun run() {
        api = DiscordApiBuilder().setToken(token).login().join()
        println("Logged in as ${api.yourself.discriminatedName}")

        SlashCommand.with("ping", "Pings the bot").createGlobal(api).join()
        api.addSlashCommandCreateListener(Ping(this))
    }
}